Userland/Libraries/LibWeb/ServiceWorker/Job.cpp

```diff
@@ -135,13 +144,284 @@ static void register_(JS::VM& vm, JS::NonnullGCPtr<Job> job)
     update(vm, job);
 }
 
+// Used to share internal Update algorithm state b/w fetch callbacks
+class UpdateAlgorithmState : JS::Cell {
+    JS_CELL(UpdateAlgorithmState, JS::Cell);
+
+public:
+    static JS::NonnullGCPtr<UpdateAlgorithmState> create(JS::VM& vm)
+    {
+        return vm.heap().allocate_without_realm<UpdateAlgorithmState>();
+    }
+
+    OrderedHashMap<URL::URL, JS::NonnullGCPtr<Fetch::Infrastructure::Response>>& updated_resource_map() { return m_map; }
+    bool has_updated_resources() const { return m_has_updated_resources; }
+    void set_has_updated_resources(bool b) { m_has_updated_resources = b; }
+
+private:
+    UpdateAlgorithmState() = default;
+
+    virtual void visit_edges(JS::Cell::Visitor& visitor) override
+    {
+        Base::visit_edges(visitor);
+        for (auto& it : m_map)
+            visitor.visit(it.value);
```

Comment:
Can this be `visitor.visit(m_map);`?

Reply:
Well how about that, it can! I didn't know we'd added those template overloads.